helix-core/src/main/java/org/apache/helix/messaging/handling/HelixStateTransitionHandler.java

```diff
@@ -227,56 +172,73 @@ void postHandleMessage() {
         rec.getMapFields().put(partitionKey, null);
         ZNRecordDelta delta = new ZNRecordDelta(rec, MergeOperation.SUBTRACT);
 
-        List<ZNRecordDelta> deltaList = new ArrayList<ZNRecordDelta>();
+        List<ZNRecordDelta> deltaList = new ArrayList<>();
         deltaList.add(delta);
         _currentStateDelta.setDeltaList(deltaList);
-        _stateModelFactory.removeStateModel(resource, partitionKey);
-      } else {
-        // if the partition is not to be dropped, update _stateModel to the TO_STATE
+        _stateModelFactory.removeStateModel(_message.getResourceName(), partitionKey);
+      }
+      // if the partition is not to be dropped, update _stateModel to the TO_STATE
```

Comment:
The comment seems not to match the code logic here.

Reply:
TFTR. Updated.